core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2385,16 +2385,24 @@ private static class FloorImplementor extends MethodNameImplementor {
         case MONTH:
         case WEEK:
         case DAY:
+        case DECADE:
+        case CENTURY:
+        case MILLENNIUM:
           final Expression dayOperand0 =
               preFloor ? call(operand0, type, TimeUnit.DAY) : operand0;
           return Expressions.call(floorMethod,
               translator.getLiteral(operand1), dayOperand0);
         default:
+          if (call.op.getName().equals("DATE_TRUNC")) {
```

Comment:
Why this behavior for `DATE_TRUNC` and not other functions? (e.g. `TIMESTAMP_TRUNC`, `DATETIME_TRUNC`)

Is it possible to add a test for this behavior?

Probably good to add `SqlKind.DATE_TRUNC`. Switching on function name is not maintainable.

Reply:
I haven't tested yet the behavior of TIMESTAMP functions.
This is strictly addressing CALCITE-5761.
It is possible that the timestamp functions have problems too.
Are you suggesting to add SqlKind.DATE_TRUNC in this PR, or should I file an issue for that?

Reply:
> I haven't tested yet the behavior of TIMESTAMP functions.

The other functions seem so closely related, it's strange to not do them.

> Are you suggesting to add SqlKind.DATE_TRUNC in this PR, or should I file an issue for that?

Yes, add SqlKind.DATE_TRUNC in this PR. Clean up tech debt as you create it.